packages/@aws-cdk/aws-ecs/lib/proxy-configuration/app-mesh-proxy-configuration.ts

```diff
@@ -0,0 +1,145 @@
+   */
+  readonly ignoredGID?: number;
+
+  /**
+   * The list of ports that the application uses.
+   * Network traffic to these ports is forwarded to the ProxyIngressPort and ProxyEgressPort.
+   */
+  readonly appPorts: number[];
+
+  /**
+   * Specifies the port that incoming traffic to the AppPorts is directed to.
+   */
+  readonly proxyIngressPort: number;
+
+  /**
+   * Specifies the port that outgoing traffic from the AppPorts is directed to.
+   */
+  readonly proxyEgressPort: number;
+
+  /**
+   * The egress traffic going to these specified ports is ignored and not redirected to the ProxyEgressPort. It can be an empty list.
+   */
+  readonly egressIgnoredPorts?: number[];
+
+  /**
+   * The egress traffic going to these specified IP addresses is ignored and not redirected to the ProxyEgressPort. It can be an empty list.
+   */
+  readonly egressIgnoredIPs?: string[];
+}
+
+/**
+ * The configuration to use when setting an App Mesh proxy configuration.
+ */
+export interface AppMeshProxyConfigurationConfigProps {
+  /**
+   * The name of the container that will serve as the App Mesh proxy.
+   */
+  readonly containerName: string;
+
+  /**
+   * The set of network configuration parameters to provide the Container Network Interface (CNI) plugin.
+   */
+  readonly properties?: AppMeshProxyConfigurationProps;
+}
+
+/**
+ * The class for App Mesh proxy configurations.
+ *
+ * For tasks using the EC2 launch type, the container instances require at least version 1.26.0 of the container agent and at least version
+ * 1.26.0-1 of the ecs-init package to enable a proxy configuration. If your container instances are launched from the Amazon ECS-optimized
+ * AMI version 20190301 or later, then they contain the required versions of the container agent and ecs-init.
+ * For more information, see [Amazon ECS-optimized AMIs](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-optimized_AMI.html).
+ *
+ * For tasks using the Fargate launch type, the task or service requires platform version 1.3.0 or later.
+ */
+export class AppMeshProxyConfiguration extends ProxyConfiguration {
+  /**
+   * Constructs a new instance of the AppMeshProxyConfiguration class.
+   */
+  constructor(private readonly props: AppMeshProxyConfigurationConfigProps) {
+    super();
+    if (props.properties) {
+      if (!props.properties.ignoredUID && !props.properties.ignoredGID) {
+        throw new Error("Either ignoredUID or ignoredGID should be specified.");
+      }
+    }
+  }
+
+  /**
+   * Called when the proxy configuration is configured on a task definition.
+   */
+  public bind(): CfnTaskDefinition.ProxyConfigurationProperty {
+    const configProps = this.props.properties;
+    const configType = "APPMESH";
+    if (configProps) {
+      return {
+        containerName: this.props.containerName,
+        proxyConfigurationProperties: removePropsEmpty([
```

Comment:
I think according to the doc you referred to and this [doc](https://docs.aws.amazon.com/app-mesh/latest/userguide/mesh-getting-started-ecs.html), at least EgressIgnoredPorts/IP can be empty and users do not have to define that. According to my experiment, I tried set EgressIgnoredPorts and IP as `undefined` and it worked well.

Reply:
Are you able to create a stack and use AppMesh?

Reply:
You cannot call removePropsEmpty if those properties need to be populated.

Reply:
Actually "AppPorts", "ProxyIngressPort", "ProxyEgressPort" these three properties are required for users, so that they cannot be empty and won't be removed by `removePropsEmpty`. As for "EgressIgnoredPorts", "EgressIgnoredIPs" after talking to agent teams, they looked at the agent code and verified that those two fields are not required and they don't do any validation for their existence.